docs/source/developer/contributing.rst

```diff
@@ -76,8 +83,87 @@ bot <https://github.com/apps/lock>`__ will lock the issue. If additional
+your changes can be merged in without delay.
+
+Definition of Ready
+^^^^^^^^^^^^^^^^^^^
+
+One of the main goals of triage is to get issues into a state where they
+are **ready** for someone to work on. Once a triager is satisfies that an 
+issue meets the definition below, they will remove the ``status:Needs Triage``
+label from it. We will not merge a pull request for an issue that still 
+needs triage.
+
+Triagers should also ensure that the issue has appropriate labels that 
+describe it, such as labels with the ``pkg:`` prefix for issues that 
+affect one or more packages.
+
+**All requested information, where applicable, is provided.** From the 
+templates in JupyterLab’s issues:
+
+For a **bug**:
+
+* Description, preferably including screen shots
+* Steps to reproduce
+* Expected behavior
+* Context: OS, browser, JupyterLab version, relevant output
+
+For a **feature request**:
+
+* Description of the problem
+* Description of the proposed solution
+* Additional context
+
+**The issue should represent real, relevant, feasible work**. In short, if a 
+knowledgeable person were to be assigned this issue, they would be able to
+complete it with a reasonable amount of effort and assistance, and it
+furthers the goals of the Jupyter project.
+
+* Issues should be unique; triage is the best time to identify duplicates.
+* Bugs represent valid expectations for use of Jupyter products and services.
+* Expectations for security, performance, accessibility, and localization match
+  generally-accepted norms in the community that uses Jupyter products.
+* The issue represents work that one developer can commit to owning, even if 
+  they collaborate with other developers for feedback. Excessively large issues 
+  should be split into multiple issues, each triaged individually, or into 
+  `team-compass <https://github.com/jupyterlab/team-compass>` issues to discuss
+  more substantive changes.
+
+Labels Used by Triagers
+^^^^^^^^^^^^^^^^^^^^^^^
+
+All new bugs and enhancement requests have the ``status:Needs Triage`` label.
+
+On a regular basis, Jupyter contributors (triage reviewers or triagers)
+review JupyterLab issues tagged
+with ``status:Needs Triage``, starting with the oldest, and determine 
+whether they meet the definition of ready.
+
+Once triaged, if the issue is ready, the reviewer removes the 
+``status:Needs Triage`` label; no additional label is required. If there 
+is not enough information in the issue as filed, the triage reviewer applies
+the ``status:Needs Info`` label and leaves ``status:Needs Triage`` in place.
+If an issue has remained in ``status:Needs Info`` for more than 14 days 
+without any follow-up communication, the reviewer should apply 
+``status:Blocked``. A blocked issue should be closed after another 14 days
+pass without a reply that unblocks it.
+
+Our expectation is that every new issue should be examined within a week of
+its creation.
```

Comment:
Do we really want to commit to a time constrain?

Reply:
I used the word "expectation" rather than "guarantee" here. I don't want us to create bureaucracy to punish ourselves. At the same time, I want to encourage people to file new issues by setting an expectation that they will be triaged reasonably quickly. I'm not sure whether a week is reasonable for us to say, but I'd prefer to say something more explicit than "reasonably quickly" in our docs.